### PR TITLE
feat/TE-7 - improved objects API

### PR DESCRIPTION
**Pull request by sergio-ivanuzzo**
* Added typed accessors for commonly used WotLK object data, so downstream code no longer needs to work directly with `UnitField`, `PlayerField`, `GameObjectField` and `FieldValue`. For example: `object.as_unit()?.health()`, `level()`, `target_guid()`, `faction_template()`, player race/class/gender, game-object state/type/flags, position and facing.
* Added an object lifecycle registry keyed by `PackedGuid`, tracking when an object was created, last updated and removed, including whether it was removed by `SMSG_DESTROY_OBJECT` or `OutOfRangeObjects`.
* Fixed `OutOfRangeObjects` handling so out-of-range GUIDs are actually removed from `ObjectMap`.
* Fixed partial update merging so fields such as powers, visible items and other array/custom update fields keep previously known values when only part of the field is present in a new update.
* Added tests for typed accessors, lifecycle tracking, object removal/recreation, movement updates, partial updates and edge cases.

### CODE REVIEW

File: src/plugins/wow/wotlk/realm/object/update_object.rs [hunk 1]
```diff
@@ -94,40 +95,123 @@ impl PacketHandler for Handler {
...
+                    if let Some(object) = objects.get_mut(&guid) {
+                        object.update(update_data);
+                        lifecycle_mutations.push(LifecycleMutation::Updated(guid));
                     }
-                }),
-            ))]));
+                }
+                ObjectMutation::Movement(guid, movement) => {
+                    if let Some(object) = objects.get_mut(&guid) {
+                        object.movement = Some(movement);
```
[the remark this answers is not in the document]
Comment (sergio-ivanuzzo) [reply src/plugins/wow/wotlk/realm/object/update_object.rs]: @codex review again plz

File: src/plugins/wow/wotlk/realm/object/accessors.rs [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+    /// Best currently known world-space position for this object.
+    #[inline]
+    pub fn position(&self) -> Option<Point3D> {
+        let movement = self.movement.as_ref()?;
+
+        if let Some(info) = movement.movement_info.as_ref() {
+            return Some(info.location.point);
+        }
+        if let Some(info) = movement.position_info.as_ref() {
+            return Some(info.world_object_point);
```
[the remark this answers is not in the document]
Comment (sergio-ivanuzzo) [reply src/plugins/wow/wotlk/realm/object/accessors.rs]: @codex review again plz
